plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java

```diff
@@ -1405,6 +1405,15 @@ public void testNativeQuerySelectFromNation()
                 .isNotFullyPushedDown(FilterNode.class);
     }
 
+    @Test
+    public void testNativeQueryColumnAlias()
+    {
+        assertThat(query("SELECT * FROM TABLE(system.query(query => 'SELECT name AS region_name FROM tpch.region WHERE regionkey = 0 ALLOW FILTERING'))"))
+                .skippingTypesCheck()
+                .hasOutputNames(ImmutableList.of("region_name"))
```

Comment:
There's no guarantee that the outputs of the query will match the column names in the table function. If you want to be absolutely sure, you should do something like:

```
SELECT region_name FROM TABLE(...)
```

which should fail if the column is not exposed by the table function.

Reply:
Changing to the suggested query hides the actual output column names. For instance, the below test pass in H2. 
```java
```

Reply:
Not sure what you mean. If the table function doesn’t expose the name, the outer query should fail with a column not found error.

Also, what do you mean by “passes on H2”? That query should not even run against H2, since it H2 doesn’t support table functions, or any of the table functions in Trino.

Reply:
For example, given:

```
SELECT region_name FROM TABLE(system.query(query => 'SELECT name FROM tpch.region WHERE regionkey = 0'));
```

=>

```
Query 20230225_223332_00030_tgtxv failed: line 1:8: Column 'region_name' cannot be resolved
```

Reply:
We have `TestJdbcConnectorTest` running on H2. 

The given example ensures column names **_without case sensitivity_**. The below query (added a column alias to the table function) should pass in my understanding because `SELECT name FROM (SELECT name as "NAME" FROM region)` is a valid query in Trino. This is what happening on the above H2 example, the table function exposes the column names with uppercase, but adding the projection lowercase the final output column names. That's why I didn't add the projection. 
```
SELECT name FROM TABLE(system.query(query => 'SELECT name AS "NAME" FROM tpch.region WHERE regionkey = 0'));
  name
--------
 AFRICA
```

Reply:
> We have TestJdbcConnectorTest running on H2.

Some tests do, but QueryAssert does not use H2. It runs the given query and compares the results to the output of the .matches() clause (which is another query that runs in Trino).

Reply:
> The given example ensures column names without case sensitivity.

Well, it tests that the name is resolvable using Trino's identifier resolution rules (which, unfortunately, currently don't follow proper SQL semantics).

Reply:
> Some tests do, but QueryAssert does not use H2. It runs the given query and compares the results to the output of the .matches() clause (which is another query that runs in Trino).

`TestJdbcConnectorTest` uses `createH2QueryRunner`. The behind database is H2 regardless of assertion methods.

Reply:
Added some test cases and comments to clarify the above behaviors.

Reply:
@martint Gentle reminder.